extensions/folder-search/src/search.tsx

```diff
@@ -55,6 +56,14 @@ export default function Command() {
   const [isShowingDetail, setIsShowingDetail] = useState<boolean>(true);
   const [results, setResults] = useState<SpotlightSearchResult[]>([]);
 
+  // hack to fix annoying double text during fallback. Typing helloworld results in helloworldhelloworld
```

Comment:
I also found this bug, but don't know how to reproduce it stably. If you know, I can check if it is a problem caused by the React component state.

Reply:
The callback parameter below looks a bit suspicious, as it will cause searchSpotlight to be executed every time the UI is updated.

```
usePromise(
    searchSpotlight,
    [
      searchText,
      searchScope,
      abortable,
      (result: SpotlightSearchResult) => {
        setResults((results) => [result, ...results].sort(lastUsedSort));
      },
    ],
   ...
)
```
--->
```
const callback = useCallback(
  (result: SpotlightSearchResult) => {
        setResults((results) => [result, ...results].sort(lastUsedSort));
   },
)

usePromise(
    searchSpotlight,
    [
      searchText,
      searchScope,
      abortable,
      callback
    ],
   ...
)
```

Reply:
That is the original code though. I just patched it up to fix these two bugs.

Reply:
Is this something we could patch too, sounds like something that would be a good idea?

Reply:
Oh I was looking at this on my phone and did not realize there was a proposed change from @fengtianze. Testing this now and if it works I'll update the PR.

Reply:
No luck with this code. Note the code as is did not work:

  ```
const searchCallback = useCallback(
    (result: SpotlightSearchResult) => {
          setResults((results) => [result, ...results].sort(lastUsedSort));
     },
     [] //I needed to add this as I was getting an error
  )
```

Reply:
I just tried debugging, but couldn't reproduce the issue. (I've encountered this a few times before.)

I found that the rendering performance of this extension has optimization space, it re-renders the UI every time it searches a directory. Sadly, I am a frontend developer and I don't know how to optimize `node-spotlight`.